### PR TITLE
python37Packages.tensorflow: 1.15.2 Fix

### PR DESCRIPTION
**Pull request by Rakesh4G**
###### Motivation for this change
Fixing Broken python37Packages.tensorflow: 1.15.2 since 
https://github.com/NixOS/nixpkgs/commit/377324ca6d05eaf76788ff95f17bcb6895bdcf10

As per https://github.com/tensorflow/tensorflow/issues/41061#issuecomment-662222308

Numpy < 1.19 should fix the issue

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by danieldk** [comment]
Jikes. Maybe we should consider removing Tensorflow 1? It seems like even keeping Tensorflow 2 working is quite a burden.

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -5095,6 +5095,8 @@ in {
     else
       callPackage ../development/python-modules/numpy { };
 
+  numpy_1_18_5 = callPackage ../development/python-modules/numpy/1.18.5.nix { }
```
Comment (danieldk) [inline pkgs/top-level/python-packages.nix:5098]: Missing semicolon breaks the build.

```suggestion
  numpy_1_18_5 = callPackage ../development/python-modules/numpy/1.18.5.nix { };
```

**Review by danieldk — CHANGES_REQUESTED** [review]

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (Rakesh4G) [reply pkgs/top-level/python-packages.nix]: Done

**Review by FRidh — CHANGES_REQUESTED** [review]
Having another version of numpy is not an option. 

In my opinion this expression needs to be removed. If you or anyone else needs older versions around, then you need to maintain them yourselves elsewhere.

**Comment by jonringer** [comment]
> If you or anyone else needs older versions around, then you need to maintain them yourselves elsewhere.

That will likely need to be the path we go. 

One option off the top of my head would be to have a top_level attr, then apply overlays so that there's valid python package sets for those packages:

```
  tensorflow_1.python37Packages.tensorflow
```

however, this will likely lead to confusion as picking packages between `tensorflow_1.python37Packages` and `pkgs.python37Packages` will likely lead to runtime errors. Also, the python package set in it's current form doesn't allow for many overlays to be composed, so users wouldn't be able to apply their own overlays

**Comment by CMCDragonkai** [comment]
Can you leave the breaking of compatibility (i.e. removing TF1) after 20.09 is released.

**Comment by danieldk** [comment]
> Can you leave the breaking of compatibility (i.e. removing TF1) after 20.09 is released.

The build has been broken for ~1.5 month, so I am not sure what value providing the Tensorflow 1 derivation as-is has (I propose to mark it broken now, see #96288 ). Also that NixOS 20.03 also has 1.15.2 and it build succesfully:

https://hydra.nixos.org/build/125764352

So people who need it, but do not want to maintain their own Tensorflow 1 derivation can always compose a Python environment from release-20.03.

**Comment by Rakesh4G** [comment]
> Having another version of numpy is not an option.
> 
> In my opinion this expression needs to be removed. If you or anyone else needs older versions around, then you need to maintain them yourselves elsewhere.

@FRidh 

I have tried to move numpy override inside tensorflow itself. I hope this will resolve this concern. 
@CMCDragonkai

**Comment by danieldk** [comment]
> I have tried to move numpy override inside tensorflow itself. I hope this will resolve this concern.

The issue is that if somebody uses the tensorflow *and* numpy, that two versions of numpy end up in Python's path.

**Comment by FRidh** [comment]
Like @danieldk said. This is simply not going to work and we need to remove 1.15.

**Comment by Rakesh4G** [comment]
ok.

**Pull request closed without merging**